### PR TITLE
Cleaning up the FlowcellRunMetricsParser module and adjust to X10

### PR DESCRIPTION
**Pull request by mayabrandi**
This will also make it more easy to adjust the code so that it later can handle the X10 html output files. 

The _html_tables_to_lists_of_tuples() function can pars both Demultiplex_Stats.htm files generated by HiSeq/MiSeq and laneBarcode.html files generated by the new X10 machines.

### CODE REVIEW

**Comment by vezzi** [comment]
@mayabrandi are we sure that removing `parse_demultiplex_stats_htm` does not affect things running on production.... It is not used in any other place now. @guillermo-carrasco maybe knows....

File: scilifelab_parsers/qc/qc.py [hunk 1]
```diff
@@ -129,78 +129,34 @@ def by_lane_yield(data):
...
+        metrics = {}
+        keys = []
+        header = None
+        for table in bs.findAll('table'):
+            header = table.findPrevious('h2')
+            if header:
+                header = header.text.replace(' ','_')
+            if header not in metrics.keys():
+                metrics[header] = []
+            if not metrics[header]:
```
Comment (senthil10) [inline scilifelab_parsers/qc/qc.py]: cat we merge both if loops ? I mean if this `if not metrics[header]` is necessary at all, as there is no any `else` block ?

Comment (mayabrandi) [inline scilifelab_parsers/qc/qc.py]: Yes its super needed @senthil10 :) 

I only want to add content to metrics[header] if its still an empty list. The loop will find the "header" sevelal rounds, but i only want to add content the first time table.findChildren('tr') gives something.

**Comment by mayabrandi** [comment]
Yes. The function parse_demultiplex_stats_htm  is not removed, Its just rewritten and compressed, and gives the same output as before. So no worries. https://github.com/vezzi

File: scilifelab_parsers/qc/qc.py (hunk 1 above)
Comment (mayabrandi) [inline scilifelab_parsers/qc/qc.py]: I added some comments to the code to make it more clear. Can we merge now? @senthil10

**Comment by senthil10** [comment]
hey @mayabrandi , I am happy to merge :) 
@vezzi : Is that okay ?

**Pull request merged**